EventAggregator.Tests/EventAggregatorTests.cs

```diff
@@ -164,17 +268,479 @@ public void Should_throw_when_listener_with_no_interfaces_added()
             typeof(ArgumentException).ShouldBeThrownBy(() => eventAggregator.AddListener(noListenerInterfaces, null));
         }
 
+#if ASYNC
+        [Fact]
+        public void Should_send_message_when_mixing_async_nonasync_listners()
+        {
+            var someMessageHandler = new SomeMessageHandler();
+            var someMessageHandlerAsync = new SomeMessageHandlerAsync();
+            var eventAggregator = new EventAggregator();
+
+            eventAggregator.AddListener(someMessageHandler);
+            eventAggregator.AddListener(someMessageHandlerAsync);
+            eventAggregator.SendMessage<SomeMessage>();
+            someMessageHandler.EventsTrapped.Count().ShouldEqual(1);
+            someMessageHandlerAsync.EventsTrapped.Count().ShouldEqual(1);
+        }
+
+        [Fact]
+        public async Task Should_send_message_async()
+        {
+            var someMessageHandler = new SomeMessageHandlerAsync();
+            var eventAggregator = new EventAggregator();
+
+            eventAggregator.AddListener(someMessageHandler);
+            await eventAggregator.SendMessageAsync<SomeMessage>();
+            someMessageHandler.EventsTrapped.Count().ShouldEqual(1);
+        }
+#if UNWRAP_EX
+        [Fact]
+        public async Task Should_throws_invalid_operation_exception_when_send_async_to_sync_listener()

```

Comment:
Actually there is an async to sync dispatcher internally. This Fact throws because the sync listener throws an exception. So the problem was in the method name that I fixed by renaming it.